### PR TITLE
Rust 1.71.1

### PR DESCRIPTION
**Pull request by tranzystorek-io**
- rust-bootstrap: update to 1.71.1
- cargo-bootstrap: update to 1.71.1
- rust: update to 1.71.1
- cargo: update to 1.71.1

closes https://github.com/void-linux/void-packages/pull/45198

#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by classabbyamp** [comment]
should probably include https://github.com/void-linux/void-packages/pull/45198 here

**Comment by tranzystorek-io** [comment]
Results of rebuild of `build_style=cargo` packages on x86_64 (commit [4513be4](https://github.com/void-linux/void-packages/pull/45461/commits/4513be44818abbc8765df30310fad2572e386737)):

**broken on master**

- pijul

**bad checksum**

- peshming

(No regressions from 1.71.0)

**Comment by tranzystorek-io** [comment]
added the gcc depends change as well

File: srcpkgs/cargo/template [hunk 1]
```diff
@@ -11,8 +11,8 @@ short_desc="Rust package manager"
 maintainer="Enno Boland <gottox@voidlinux.org>"
 license="MIT, Apache-2.0"
 homepage="https://crates.io/"
-distfiles="https://github.com/rust-lang/cargo/archive/refs/tags/${_cargo_revision}.tar.gz"
-checksum=c0c15a35852c114859cfe11c5c76ee12105d66d7c90e650af83ac7b8a6a856fc
+distfiles="https://github.com/rust-lang/cargo/archive/${_cargo_commit}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/cargo/template:14]: https://github.com/rust-lang/cargo/issues/12460 your tag is there now :)

Comment (tranzystorek-io) [reply srcpkgs/cargo/template]: Thanks!

**Pull request merged**